x-pack/plugins/security_solution/public/detections/hooks/trigger_actions_alert_table/use_actions_column.tsx

```diff
@@ -42,6 +43,13 @@ export const getUseActionColumnHook =
       } = getAlertsDefaultModel(license),
     } = useSelector((state: State) => eventsViewerSelector(state, tableId));
 
+    const { loading: listsConfigLoading, needsConfiguration: needsListsConfiguration } =
+      useListsConfig();
```

Comment:
@yctercero @kqualters-elastic This call to `useListsConfig` is  the reason for the second call in the screenshot. I'm not sure if there's a way to avoid this.

Reply:
I am wondering if using `useQuery` with a queryKey will make all of our problem going away.

Reply:
that would probably work, but think it would be better to just have this check made once, by moving this logic way above any components that get rendered as an array, like row_actions. about half of the changes in this pr are passing the prop hasAccessToLists down, could we perhaps avoid that by storing it in a context or redux? would greatly simplify the pr and ensure this isn't likely to happen again imo

Reply:
Thanks for the suggestion @kqualters-elastic. Are you suggesting using a context instead of the `useListsConfig` hook? Or just that, we use the context instead of the prop changes in this PR. If the latter, would this mean that we have two sets of values for `api/lists/index`? One via a context value and another via `useListsConfig`? I'm not very familiar with this are of the code which is why I went with this approach of passing down values.

cc @yctercero @rylnd

Reply:
@ashokaditya I think the suggestion is to define `hasAccessToLists` as a hook based on `useContext` and `useListsConfig`.